Added harvest_craft_poedb_praser

Kuon-dev opened the pull request:
Added the parser to scrape data from poedb of harvest seed crafts.

Kuon-dev committed:
Added harvest_craft_poedb_praser

Added the parser to scrape data from poedb of harvest seed crafts

namlehong commented:
some code review
- input of harvest_craft_poe_db_parser should be html_text to improve in the unit test process. and you could create another function named get_harvest_craft_html() and return html text
- return parsed data rather than direct write to json file
- try using yield to improve your code and performance

Kuon-dev committed:
Updated parser.py

Remove unnecessary codes and process, changed the return function to return the dict.

Code has passed test case.

Added an option to dump data into a json file, if the json file is empty. Else will let the user know the json has contents in it.

Kuon-dev commented:
Code has changed to return a list of parsed data. The test case was successful.

Might need some assistance / reference on using the yield function as I don't have much clue on that part unfortunately.
